riscv/mmu.cc

```diff
@@ -258,7 +275,7 @@ reg_t mmu_t::pmp_homogeneous(reg_t addr, reg_t len)
   return true;
 }
 
-reg_t mmu_t::s2xlate(reg_t gva, reg_t gpa, access_type type, access_type trap_type, bool virt, bool hlvx)
+reg_t mmu_t::s2xlate(reg_t gva, reg_t gpa, int& type, int trap_type, bool virt, bool hlvx)
```

Comment:
Why use a reference for type?

Reply:
Nevermind, I see why. This smells bad but I can't think of a better way. :(

Reply:
This is going to lead to a lot of confusion.  Return an aggregate (e.g. `struct { reg_t addr; int type; };`) instead?

Reply:
OK. I'll try this way.